chain/ethereum/src/network_indexer/metrics.rs

```diff
@@ -49,246 +48,152 @@ pub struct NetworkIndexerMetrics {
 
 impl NetworkIndexerMetrics {
     pub fn new(
-        subgraph_id: SubgraphDeploymentId,
+        subgraph_id: &SubgraphDeploymentId,
         stopwatch: StopwatchMetrics,
         registry: Arc<dyn MetricsRegistry>,
     ) -> Self {
+        let make_gauge = |name: &str, help: &str| {
+            registry
+                .new_deployment_gauge(name, help, subgraph_id.as_str())
+                .expect(
+                    format!("failed to register metric `{}` for {}", name, subgraph_id).as_str(),
+                )
+        };
+
+        let make_counter = |name: &str, help: &str| {
+            registry
+                .new_deployment_counter(name, help, subgraph_id.as_str())
+                .expect(
+                    format!("failed to register metric `{}` for {}", name, subgraph_id).as_str(),
+                )
+        };
+
         Self {
             stopwatch,
 
-            chain_head: registry
-                .new_gauge(
-                    format!("{}_chain_head", subgraph_id),
-                    "The current chain head block".into(),
-                    HashMap::new(),
-                )
-                .expect(format!("failed to register metric `{}_chain_head`", subgraph_id).as_str()),
+            chain_head: make_gauge("deployment_chain_head", "The current chain head block"),
 
-            local_head: registry
-                .new_gauge(
-                    format!("{}_local_head", subgraph_id),
-                    "The current local head block".into(),
-                    HashMap::new(),
-                )
-                .expect(format!("failed to register metric `{}_local_head`", subgraph_id).as_str()),
+            local_head: make_gauge("deployment_local_head", "The current local head block"),
```

Comment:
On these two (and others related to the subgraph network): it might make sense to add a `network` label in order to be future proof for the days where one subgraph indexes across multiple networks. But for now, this is fine.